### PR TITLE
Issue 10 implement route protection for user and story using middleware

### PR DESCRIPTION
**Pull request by NirjoyDebnath**

### CODE REVIEW

File: src/services/DTOs/authDTO.ts [hunk 1]
```diff
@@ -21,6 +23,7 @@ export class SignUpUserDTO implements ISignUpUserDTO {
     this.Name = signUpUserInput.Name;
     this.Role = Role.user;
     this.JoinDate = new Date();
+    this.dataValidate();
```
Comment (Propo41) [inline src/services/DTOs/authDTO.ts:26]: method name should be a verb. better would be to use validateData() instead

File: src/routes/storyRoute.ts [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
-  .patch(storyController.updateStoryById)
-  .delete(storyController.deleteStoryById);
+  .patch(
+    authMiddleware.authenticateUser,
+    storyMiddleware.authorizedForUpdate,
+    storyController.updateStoryById
+  )
+  .delete(
+    authMiddleware.authenticateUser,
+    storyMiddleware.authorizedForDelete,
```
Comment (Propo41) [inline src/routes/storyRoute.ts:22]: better name would be `authorizeDeletion()`

**Review by Propo41 — APPROVED** [review]

File: src/middlewares/authMiddleware.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      ENV.SecretKey
+    ) as ITokenInfo;
+    const auth: IAuth | undefined = await getAuthByUserName(tokenInfo.userName);
+    if (!auth) {
+      return next(new AppError(400, 'User doesnot exist'));
+    }
+    const passwordModifiedAtInSecond: number =
+      auth.PasswordModifiedAt.getTime() / 1000;
+    if (passwordModifiedAtInSecond > tokenInfo.iat) {
+      return next(new AppError(401, 'You are not authorized'));
```
Comment (edge555) [inline src/middlewares/authMiddleware.ts:32]: Message should be token expired, Please log in again.
Also add another condition When Current time > Token expire time

File: src/routes/authRoute.ts [hunk 4]
```diff
@@ -1,11 +1,13 @@
...
 
 const router = express.Router();
 
-router.route('/signUp').post(signUp);
-
-router.route('/logIn').post(logIn);
+router.route('/signUp').post(authController.signUp);
+router.route('/logIn').post(authController.logIn);
+router
+  .route('/updatePassword')
```
Comment (edge555) [inline src/routes/authRoute.ts:10]: updatePassword should not be seperate route, It will be inside user/:id patch

**Pull request merged**